File: benchmarks/solutions/ruby/r2-json-transform.rb
require "json"

begin
  users = JSON.parse($stdin.read)
rescue JSON::ParserError => e
  $stderr.puts "error: #{e.message}"
  exit 1
end

result = users
  .select { |u| u["active"] && u["age"] > 18 }
  .map { |u| { name: u["name"], email: u["email"] } }

puts JSON.generate(result)

File: benchmarks/solutions/python/r2-json-transform.py
import json
import sys

try:
    users = json.loads(sys.stdin.read())
except json.JSONDecodeError as e:
    print(f"error: {e}", file=sys.stderr)
    sys.exit(1)

result = [
    {"name": u["name"], "email": u["email"]}
    for u in users
    if u["active"] and u["age"] > 18
]
print(json.dumps(result))
